JavaScript development/Frameworks/React.md

```diff
@@ -0,0 +1,389 @@
+
+### File organization and naming
+- as it is a good practice to divide React components as controllers and presentational components, you should place them in separate folders (example: **containers** and **components**)
+
+```
+...
+├── components
+├── containers 
+...
+```
+
+- each component should have it's own folder containing it's logic in a JSX file and it's stylesheet file.
+ (If the app is (or becomes) more complex the components can be placed in additional logically separated sub-folders for better readability and organization).
+- use **camelCase** for general folder naming and **PacalCase** for React components, their stylesheet files and their containing folders 
+- each component type's root folder should have an **index.jsx** file which exports all components, so you can later import them from a single place
```

Comment:
1. We're not using `jsx` extensions, only `js`. When using typescript, we have to use `tsx` (compiler imitation).
2. We're doing this on some projects, but I'm not actually convinced it's a good practice, because it makes code splitting harder/impossible. Might be better to just remove this.